controllers/space/space_controller_test.go

```diff
@@ -737,16 +737,126 @@ func TestPromoteSpace(t *testing.T) {
 		})
 	})
 
+	t.Run("tier update (same tier but updated references)", func(t *testing.T) {
+		// given that Space is set to the same tier that has been updated and the corresponding NSTemplateSet is not up-to-date
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierNameAndHashLabelFor(basicTier), // assume that at this point, the `TemplateTierHash` label was already removed by the ChangeTierRequestController
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		hostClient := test.NewFakeClient(t, s, basicTier, otherTier)
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity", nstemplatetsettest.WithReadyCondition()) // NSTemplateSet has references to old basic tier
```

Comment:
if you used references from the "older" version of the basic tier:
```
nstemplatetsettest.WithReferencesFor(olderBasicTier)
```
it would make the code more clear why the content is not matching

Reply:
That's a good suggestion. Thanks! 3a63b3ca6927a4f2ccc3943606adae44b7121c33